src/components/ReplaceServerDetails.js

```diff
@@ -66,22 +65,81 @@ class ReplaceServerDetails extends Component {
       });
   }
 
+  initInputs = () => {
+    let inputs = {};
+
+    if (this.isCompute()) {
+      MODEL_SERVER_PROPS.forEach(input_name => {
+        inputs[input_name] = '';
+      });
+    }
+    else {
+      REPLACE_SERVER_MAC_IPMI_PROPS.forEach(input_name => {
+        inputs[input_name] = '';
+      });
+    }
+
+    return Map(inputs);
+  }
+
+  initInputsValid = () => {
+    let inputValid = {};
+
+    if (this.isCompute()) {
+      MODEL_SERVER_PROPS.forEach(input_name => {
+        inputValid[input_name] = undefined;
+      });
+    }
+    else {
+      REPLACE_SERVER_MAC_IPMI_PROPS.forEach(input_name => {
+        inputValid[input_name] = undefined;
+      });
+    }
+
+    return Map(inputValid);
+  }
+
+  isServerInputsValid = () => {
+    // if it is compute node and install os is not checked
+    // only check MAC and IMPI inputs when user inputs them
+    if (this.isCompute() && !this.state.isInstallOsSelected) {
+      return this.state.isValid.every((value, key) =>{
+        if (REPLACE_SERVER_MAC_IPMI_PROPS.includes(key)) {
+          return value === true || value === undefined;
+        }
+        else {
+          return value === true;
+        }
+      });
+    }
+
+    return this.state.isValid.every((value) => value === true);
+  }
+
   isFormInputValid = () => {
 
     // The form is valid if all of the fields are valid.  The OS install password only has
     // to be valid when the OS Install checkbox is selected
-    return this.state.isValid.every((value) => value === true) &&
+    return this.isServerInputsValid() &&
       (this.state.isOsInstallPasswordValid || !this.state.isInstallOsSelected);
   }
 
   handleDone = () => {
-
     let data = {};
-
-    data['mac-addr'] = this.state.inputValue.get('mac-addr');
-    data['ilo-ip'] = this.state.inputValue.get('ilo-ip');
-    data['ilo-user'] = this.state.inputValue.get('ilo-user');
-    data['ilo-password'] = this.state.inputValue.get('ilo-password');
+    // if it is compute node, will take all user inputs
+    // for the new compute node
+    if(this.isCompute()) {
+      MODEL_SERVER_PROPS.forEach(input_name => {
+        data[input_name] = this.state.inputValue.get(input_name);
+      });
+      data['role'] = this.props.data['role'];
```

Comment:
This does not look right. The caller is supplying this data already, so we should not be turning around and feeding it right back. Let the caller insert it into this object if it is needed.

Reply:
will update